l10n_fr_siret/partner.py

```diff
@@ -58,6 +57,12 @@ def _get_siret(self):
     @api.constrains('siren', 'nic')
     def _check_siret(self):
         """Check the SIREN's and NIC's keys (last digits)"""
+        # Check unicity of siret
+        for partner in self.env['res.partner'].search([('id', '!=', self.id),('siret', 'not in', [False,''])]):
+            siret = str(self.siren) + str(self.nic)
+            if siret == partner.siret:
+                raise ValidationError(_("Siret must be unique !\n It was defined for '%s'.") % partner.name)
+
```

Comment:
why not a UNIQUE SQL index on siren + nic ? I'm concerned about performance issues with this implementation.

Reply:
Note that it is not possible to have a UNIQUE SQL index based on siren (or siren + nic) because, with the current code of l10n_fr_siret, siren + nic is a commercial field, so it is synced between parent and child partners, cf https://github.com/OCA/l10n-france/blob/10.0/l10n_fr_siret/models/partner.py#L87

I think it sucks, but it is the same behaviour as the VAT field, cf https://github.com/odoo/odoo/blob/10.0/odoo/addons/base/res/res_partner.py#L411

I would prefer to avoid this sync and oblige developers to read the "commercial_partner" to get the vat or siren/siret... but it's difficult to change this.